server/src/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -388,6 +388,19 @@ public DomainRouterVO startVirtualRouter(final DomainRouterVO router, final User
                     routerToBeAvoid = rrouter;
                 }
             }
+        } else if (router.getVpcId() != null) {
+            final List<DomainRouterVO> routerList = _routerDao.listByVpcId(router.getVpcId());
+            for (final DomainRouterVO rrouter : routerList) {
+                if (rrouter.getHostId() != null && rrouter.getIsRedundantRouter() && rrouter.getState() == State.Running) {
+                    if (routerToBeAvoid != null) {
```

Comment:
should this also check that the routerList.size() is >= 2?

Reply:
it should be checked at some other places I think.
I am too lazy I just copied the lines from 380 to 390
https://github.com/apache/cloudstack/blob/4.11/server/src/com/cloud/network/router/NetworkHelperImpl.java#L380,L390

Reply:
it should be checked at some other places I think.
I am too lazy I just copied the lines from 380 to 390
https://github.com/apache/cloudstack/blob/4.11/server/src/com/cloud/network/router/NetworkHelperImpl.java#L380,L390